spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/JobExecutionDaoContainer.java

```diff
@@ -41,7 +39,8 @@ public class JobExecutionDaoContainer {
 
 	public JobExecutionDaoContainer(DataSource dataSource, SchemaService schemaService) {
 		for (SchemaVersionTarget target : schemaService.getTargets().getSchemas()) {
-			JdbcSearchableJobExecutionDao jdbcSearchableJobExecutionDao = new JdbcSearchableJobExecutionDao();
+			BatchVersion batchVersion = target.getSchemaVersion().equals(AppBootSchemaVersion.BOOT3) ? BatchVersion.BATCH_5 : BatchVersion.BATCH_4;
```

Comment:
This ternary has surfaced in many places in this PR, I wonder if we shoudl abstract this out somewhere?